readiness/controller/pkg/readiness/readiness_controller.go

```diff
@@ -34,14 +36,142 @@ type ReadinessReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.6.4/pkg/reconcile
 func (r *ReadinessReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
```

Comment:
Context should be initialized with timeout here and passed along wherever needed. This helps in avoiding network calls that may take forever.

```
ctxCancel, cancel := context.WithTimeout(ctx, constants.ContextTimeout)
defer cancel()
```

Reply:
Done